### DISCUSSION TITLE
How to get 2d images from a loaded 3d image with a Transform ?

### DISCUSSION THREAD

**Discussion by Kazezaka** [Project-MONAI/MONAI#1983] [Q&A] [answered]
Hello,

I am currently working with 3d images, that I want to feed as 2d images to a UNET.
I was wondering if there is a transform that allows me to :

get a dataset containing N*Z images of shape (B,C,X,Y) 
instead of a dataset containg N images of shape (B,C,X,Y,Z)
when calling CacheDataset.

Or should I just load 2d images from the beginning ? (but that would be slow)
Or maybe I'm just missing something.

**Comment by Nic-Ma** [comment, marked as answer]
Hi @Kazezaka ,

Thanks for your interest here. We don't have such a CacheDataset feature so far.
Welcome to contribute a PR if you want.

Thanks.

**Reply by rijobro** [reply]
Might be able to use a Lamdba like this (untested):

```python
def split_im_to_slices(img, dim=-1):
    return np.split(img, img.shape[dim], axis=dim))

transform = Compoe([
    LoadImage...,
    Lambda(func=split_im_to_slices),
    ...,
])
```
